verticapy/tests_new/performance/vertica/test_qprof.py

```diff
@@ -0,0 +1,1384 @@
+"""
+Copyright  (c)  2018-2024 Open Text  or  one  of its
+affiliates.  Licensed  under  the   Apache  License,
+Version 2.0 (the  "License"); You  may  not use this
+file except in compliance with the License.
+You may obtain a copy of the License at:
+http://www.apache.org/licenses/LICENSE-2.0
+Unless  required  by applicable  law or  agreed to in
+writing, software  distributed  under the  License is
+distributed on an  "AS IS" BASIS,  WITHOUT WARRANTIES
+OR CONDITIONS OF ANY KIND, either express or implied.
+See the  License for the specific  language governing
+permissions and limitations under the License.
+"""
+import io
+import re
+import logging
+import json
+from contextlib import redirect_stdout
+from graphviz import Digraph
+import pandas as pd
+import numpy as np
+import pytest
+from verticapy.datasets import load_titanic
+import verticapy as vp
+from verticapy.tests_new.performance.vertica import QPROF_SQL1, QPROF_SQL2
+from verticapy.performance.vertica import QueryProfiler
+from verticapy.connection import current_cursor
+from verticapy.core.vdataframe import vDataFrame
+
+pd.set_option("display.max_columns", None)
+pd.set_option("display.max_rows", None)
+logger = logging.getLogger(__name__)
+
+
+class TestQueryProfiler:
+    """
+    test class for QueryProfiler
+    """
+
+    _target_tables = [
+        "dc_requests_issued",
+        "dc_query_executions",
+        "dc_explain_plans",
+        "execution_engine_profiles",
+        "query_plan_profiles",
+        "query_profiles",
+        "resource_pool_status",
+        "host_resources",
+    ]
+
+    @pytest.fixture(name="qprof_data", scope="class")
+    def data_setup(self, titanic_vd, schema_loader):
+        """
+        test function for data setup
+        """
+        # titanic = titanic_vd.to_pandas()
+
+        transactions = []
+        qdurations = []
+        query_label_map = {
+            "qprof_sql1": "QueryProfiler_sql1_requests_UT",
+            "qprof_sql2": "QueryProfiler_sql2_requests_UT",
+            "qprof_sql3": "QueryProfiler_sql3_requests_UT",
+        }
+
+        qprof_sql3 = f"""SELECT /*+LABEL('QueryProfiler_sql3_requests_UT')*/ ticket, substr(ticket, 1, 5) AS ticket, AVG(age) AS avg_age FROM {schema_loader}.titanic GROUP BY 1"""
+
+        queries = [QPROF_SQL1, QPROF_SQL2, qprof_sql3]
+
+        for key, query in zip(query_label_map.keys(), queries):
+            current_cursor().execute(query)
+            sql = f"select transaction_id, statement_id from v_monitor.query_requests where request_label = '{query_label_map[key]}' and request not like 'PROFILE%' ORDER BY start_timestamp DESC LIMIT 1"
+            res = current_cursor().execute(sql).fetchall()
+            print(res)
```

Comment:
The `print()` statement on the line above doesn't have a lot of context. 

Does the print need to remain in the test? 

If it does, could it be a `logging.info()` or `logging.debug()` statement with some extra context about what happened?

Reply:
This print statement is just for debugging purpose.
 I will update test to use logger function.